sectionproperties/pre/sections.py

```diff
@@ -293,224 +622,757 @@ def plot_geometry(self, ax=None, pause=True, labels=False, perimeter=False):
+
+            Mesh generated from the above geometry.
         """
+        if len(mesh_sizes) == 1:
+            mesh_sizes = mesh_sizes * len(self.control_points)
+        self.mesh = pre.create_mesh(
+            self.points, self.facets, self.holes, self.control_points, mesh_sizes
+        )
+        return self
 
-        # get facet points
-        p1 = self.points[facet[0]]
-        p2 = self.points[facet[1]]
+    def shift_section(self, x_offset: float = 0, y_offset: float = 0):
+        """
+        Returns a new CompoundGeometry object translated by 'x_offset' and 'y_offset'.
 
-        # calculate distance between two points
-        return np.sqrt((p2[0] - p1[0]) ** 2 + (p2[1] - p1[1]) ** 2)
+        :param x_offset: Distance in x-direction by which to shift the geometry.
+        :type x_offset: float
+        :param y_offset: Distance in y-direction by which to shift the geometry.
+        :type y_offset: float
 
-    def calculate_perimeter(self):
-        """Calculates the perimeter of the cross-section by summing the length of all facets in the
-        ``perimeter`` class variable.
+        :return: CompoundGeometry object shifted by 'x_offset' and 'y_offset'
+        :rtype: :class:`sections.pre.sections.CompoundGeometry`
+        """
+        geoms_acc = []
+        for geom in self.geoms:
+            geoms_acc.append(geom.shift_section(x_offset=x_offset, y_offset=y_offset))
+        new_geom = CompoundGeometry(geoms_acc)
+        return new_geom
 
-        :return: Cross-section perimeter, returns 0 if there is no perimeter defined
-        :rtype: float
+    def rotate_section(self, angle, rot_point=None, use_radians=False):
+        """Rotates the compound geometry and specified angle about a point. If the rotation point is not
+        provided, rotates the section about the center of the compound geometry's bounding box.
+
+        :param float angle: Angle (degrees by default) by which to rotate the section. A positive angle leads
+            to a counter-clockwise rotation.
+        :param rot_point: Optional. Point *(x, y)* about which to rotate the section. If not provided, will rotate
+            about the center of the compound geometry's bounding box. Default = 'center'.
+        :type rot_point: list[float, float]
+        :param use_radians: Boolean to indicate whether 'angle' is in degrees or radians. If True, 'angle' is interpreted as radians.
+        
+        :return: CompoundGeometry object rotated by 'angle' about 'rot_point'
+        :rtype: :class:`sections.pre.sections.CompoundGeometry`
+        
+        The following example rotates a 200UB25 section clockwise by 30 degrees::
+
+            import sectionproperties.pre.sections as sections
+
+            geometry_1 = sections.i_section(d=203, b=133, t_f=7.8, t_w=5.8, r=8.9, n_r=8)
+            geometry_2 = sections.rectangular_section(d=20, b=133)
+            compound = geometry_2.align_center(geometry_1).align_top(geometry_1) + geometry_1
```

Comment:
@connorferster, update align_top as no longer exists in code

Reply:
Got 'em both now.